dev-python/wurlitzer/wurlitzer-2.0.0.ebuild

```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="Capture C-level stdout/stderr in Python "
+HOMEPAGE="https://github.com/minrk/wurlitzer"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="${PYTHON_DEPS}
+	dev-python/setuptools[${PYTHON_USEDEP}]"
+
+RDEPEND="${DEPEND}"
```

Comment:
This can all be replaced with DISTUTILS_USE_SETUPTOOLS={no,bdpend,rdepend}. Please take a look at $repo/eclass/distutils-r1.eclass.

Reply:
I don't think I understand what you mean, I added it (before inherit, as per instructions in the eclass file) and it gives me this:
```* ERROR: dev-python/wurlitzer-2.0.0::localrepo failed (depend phase):
*   Invalid DISTUTILS_USE_SETUPTOOLS={rdepend}
*
* Call stack:
*                ebuild.sh, line 616:  Called source '/var/db/repos/localrepo/dev-python/wurlitzer/wurlitzer-2.0.0.ebuild'
*   wurlitzer-2.0.0.ebuild, line  10:  Called inherit 'distutils-r1'
*                ebuild.sh, line 317:  Called __qa_source '/var/db/repos/gentoo/eclass/distutils-r1.eclass'
*                ebuild.sh, line 112:  Called source '/var/db/repos/gentoo/eclass/distutils-r1.eclass'
*      distutils-r1.eclass, line 141:  Called _distutils_set_globals
*      distutils-r1.eclass, line 129:  Called die
* The specific snippet of code:
*                      die "Invalid DISTUTILS_USE_SETUPTOOLS=${DISTUTILS_USE_SETUPTOOLS}"
*
* If you need support, post the output of `emerge --info '=dev-python/wurlitzer-2.0.0::localrepo'`,
* the complete build log and the output of `emerge -pqv '=dev-python/wurlitzer-2.0.0::localrepo'`.
* Working directory: '/usr/lib/python3.7/site-packages'
* S: '/var/tmp/portage/dev-python/wurlitzer-2.0.0/work/wurlitzer-2.0.0'```

Reply:
Ah, you do not need the {} around rdepend. Sorry, my expression was a Bash'ism of the available options.

Reply:
If I understand it correctly `DISTUTILS_USE_SETUPTOOLS=bdepend` is the default, so I don't actually have to specify this in the ebuild, right?

Reply:
No, the default is set if you set DISTUTILS_USE_SETUPTOOLS=1. It is variable that must be explicitly enabled/set.

Reply:
Are you sure? I discussed the same thing with @hartwork over at https://github.com/gentoo/gentoo/pull/14014 and he concluded:

> I like where this is going — interesting!

>This is what empty DEPEND and RDEPEND in the ebuild get me with distutils-r1:

> - BDEPEND: test? ( pytest ) + PYTHON_DEPS + setuptools
> - DEPEND: empty
> - RDEPEND: PYTHON_DEPS

>Unless setuptools is a true runtime dependency for this package — I find no imports to setuptools or pkg_resources outside of setup.py and no entrypoints in setup.py — that doesn't look too bad to me.

Reply:
DISTUTILS_USE_SETUPTOOLS was introduced in b1c6961beda1e729f3c6d301d6e0cce4de0493bf . Valid values are `no`, `bdepend` (default), `rdepend`.